front/src/modules/tasks/components/TaskRow.tsx

```diff
@@ -0,0 +1,128 @@
+import { useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+
+import { useOpenActivityRightDrawer } from '@/activities/hooks/useOpenActivityRightDrawer';
+import { CompanyChip } from '@/companies/components/CompanyChip';
+import { PersonChip } from '@/people/components/PersonChip';
+import { IconCalendar, IconComment } from '@/ui/icon';
+import {
+  Checkbox,
+  CheckboxShape,
+} from '@/ui/input/checkbox/components/Checkbox';
+import { useGetCompaniesQuery, useGetPeopleQuery } from '~/generated/graphql';
+import { getLogoUrlFromDomainName } from '~/utils';
+import { beautifyExactDate } from '~/utils/date-utils';
+
+import { TaskForList } from '../types/TaskForList';
+
+const StyledContainer = styled.div`
+  align-items: center;
+  align-self: stretch;
+  border-bottom: 1px solid ${({ theme }) => theme.border.color.light};
+  cursor: pointer;
+  display: flex;
+  height: ${({ theme }) => theme.spacing(12)};
+  padding: 0 ${({ theme }) => theme.spacing(4)};
+`;
+
+const StyledSeparator = styled.div`
+  flex: 1;
+`;
+
+const StyledTaskTitle = styled.div`
+  padding: 0 ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledCommentIcon = styled.div`
+  color: ${({ theme }) => theme.font.color.light};
+`;
+
+const StyledDueDate = styled.div`
+  color: ${({ theme }) => theme.font.color.secondary};
+  gap: ${({ theme }) => theme.spacing(1)};
+  padding-left: ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledChipsContainer = styled.div`
+  display: flex;
+  gap: ${({ theme }) => theme.spacing(1)};
+`;
+
+export function TaskRow({ task }: { task: TaskForList }) {
+  const theme = useTheme();
+  const openActivityRightDrawer = useOpenActivityRightDrawer();
+  const { data: targetPeople } = useGetPeopleQuery({
```

Comment:
This is quite poor performance-wise as there will be one call per activity.
It is not possible to fetch the entity properties directly from the graphql query since we have a morph here that forbids us to specify entity fields to retrieve.
We may fetch the People and Companies once for all tasks after we get the result of the Tasks query, by filtering ids for the union of the ids of the targets of the tasks.

Reply:
Agree!